openverse_catalog/dags/providers/provider_api_scripts/nypl.py

```diff
@@ -1,210 +1,229 @@
+        item_details = item_json.get("nyplAPI", {}).get("response")
+        if not item_details:
+            return None
         mods = item_details.get("mods")
-        title = _get_title(mods.get("titleInfo"))
-        creator = _get_creators(mods.get("name"))
-        metadata = _get_metadata(mods)
 
-        captures = item_details.get("sibling_captures", {}).get("capture", [])
-        if type(captures) is not list:
-            captures = [captures]
+        title_info = mods.get("titleInfo")
+        if isinstance(title_info, list) and len(title_info) > 0:
+            title_info = title_info[0]
+        title = title_info.get("title", {}).get("$")
 
-        _get_capture_details(
-            captures=captures, metadata=metadata, creator=creator, title=title
-        )
+        name_properties = mods.get("name")
+        creator = self._get_creators(name_properties) if name_properties else None
 
-
-def _get_capture_details(captures=None, metadata=None, creator=None, title=None):
-    if captures is None:
-        captures = []
-    for img in captures:
-        image_id = img.get("imageID", {}).get("$")
-        if image_id is None:
-            continue
-        image_url = _get_image_url(img.get("imageLinks", {}).get("imageLink", []))
-        foreign_landing_url = img.get("itemLink", {}).get("$")
-        license_url = img.get("rightsStatementURI", {}).get("$")
-        if image_url is None or foreign_landing_url is None or license_url is None:
-            continue
-
-        image_store.add_item(
-            foreign_identifier=image_id,
-            foreign_landing_url=foreign_landing_url,
-            image_url=image_url,
-            license_info=get_license_info(license_url=license_url),
-            title=title,
-            creator=creator,
-            meta_data=metadata,
+        metadata = self._get_metadata(mods)
+        category = (
+            ImageCategory.PHOTOGRAPH.value
+            if metadata.get("genre") == "Photographs"
+            else None
         )
 
-
-def _get_title(titleinfo):
-    title = None
-    if type(titleinfo) == list and len(titleinfo) > 0:
-        title = titleinfo[0].get("title", {}).get("$")
-    return title
-
-
-def _get_creators(creatorinfo):
-    if type(creatorinfo) == list:
-        primary_creator = (
-            info.get("namePart", {}).get("$")
-            for info in creatorinfo
-            if info.get("usage") == "primary"
+        captures = item_details.get("sibling_captures", {}).get("capture")
+        if not captures:
+            return None
+        if not isinstance(captures, list):
+            captures = [captures]
+        images = []
+        for img in captures:
+            image_id = img.get("imageID", {}).get("$")
+            if image_id is None:
+                continue
+            image_link = img.get("imageLinks", {}).get("imageLink", [])
+            image_url, filetype = NyplDataIngester._get_image_data(image_link)
```

Comment:
I think we can just use `self` here:
```suggestion
            image_url, filetype = self._get_image_data(image_link)
```